### PR TITLE
#31 add batch normalization

### PR DESCRIPTION
**Pull request by celina-ayl**

### CODE REVIEW

File: hub/examples/image_retraining/retrain.py [hunk 1]
```diff
@@ -745,7 +745,18 @@ def add_final_training_ops(class_count, final_tensor_name, bottleneck_tensor):
...
         with tf.compat.v1.name_scope('Wx_plus_b'):
-            logits = tf.matmul(bottleneck_input, layer_weights) + layer_biases
+            pre_activations = tf.matmul(
+                bottleneck_input,
+                layer_weights
+            ) + layer_biases
+
+            batch_norm = tf.keras.layers.BatchNormalization()(
+                pre_activations,
+                training=True
```
Comment (renney-11) [inline hub/examples/image_retraining/retrain.py:755]: i think this being hard coded can cause issues. it being hard coded means batch normalization always uses minibatch statistics, even during validation and testing, where it should use the learned running averages instead

File: hub/examples/image_retraining/retrain.py [hunk 2]
```diff
@@ -745,7 +745,18 @@ def add_final_training_ops(class_count, final_tensor_name, bottleneck_tensor):
...
+            )
+
+            logits = tf.nn.relu(batch_norm)
```
Comment (renney-11) [inline hub/examples/image_retraining/retrain.py:758]: logits is actually the output of relu(batch_norm), not raw pre-activation values. softmax_cross_entropy_with_logits expects raw logits, applying ReLU before the loss probs silently degrades training

**Review by renney-11 — CHANGES_REQUESTED** [review]
check comments, also a good read that can help: https://d2l.ai/chapter_convolutional-modern/batch-norm.html
